contents/tutorials/group-page-machine-flags.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Targeting feature flags on groups, pages, machines, and more
+date: 2023-08-08
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/tutorial-16.png
+tags: ['feature flags']
+---
+
+Most feature flags target users. To decide what value to return, PostHog’s feature flag service takes a flag key and user. What counts as a "user" or how you filter that user is up to you. Doing this enables you to target other entities like organizations, pages, machines, and more. This tutorial shows you how to target these non-user entities.
+
+## Targeting groups, teams, or organizations
+
+If you enabled [group analytics](/docs/product-analytics/group-analytics), targeting by groups, teams, or organizations is easy. When creating your feature flag, change the "Match by" value under "Release conditions" your group type name, add any conditions you want, and roll out the flag. 
+
+![Match by organization](../images/tutorials/group-page-machine-flags/org.png)
+
+For this to work, you need group analytics enabled and group identification set up. Once done, you can target your flags to specific groups.
+
+### Property or cohort filter
+
+Another way to target an organization or team is to use the email or similar repeated identifier of a user. This enables you to filter for groups without using group analytics. Similarly, you can create a cohort for the organization, then use that as a condition.
+
+![Email condition](../images/tutorials/group-page-machine-flags/email.png)
+
+## Targeting pages
+
+Another target you might want is pages. For example, to ensure certain docs pages are consistent, beta test a new pricing page, or change a page users visited many times.
+
+### Payloads
+
+You can use [feature flag payloads](/docs/feature-flags/payloads) to customize the experience depending on the page. For example, you want a button to change colors depending on the page URL for a group of users. Payloads lets you set multiple values on one feature flag which you can evaluate and use in your app.
+
+To do this, create your flag with a payload such as:
+
+```json
+{
+  "signup": "red",
+  "login": "blue",
+  "home": "green"
+}
+```
+
+In your app, get feature flag payloads, write logic to check the page URL, and use a combination of the page URL and flag payload to adjust your button. For a Next.js component, this would look like this:
```

Comment:
```suggestion
In your app, get feature flag payloads, write logic to check the page URL, and use a combination of the page URL and flag payload to adjust your button. For a react component, this would look like this:
```